### PR TITLE
Add dual-stack job for docs test

### PR DESCRIPTION
**Pull request by kfaseela**
Needed for https://github.com/istio/istio.io/pull/14257/files

### CODE REVIEW

File: prow/config/jobs/istio.io.yaml [hunk 1]
```diff
@@ -41,6 +41,16 @@ jobs:
...
 
+  - name: doc.test.dualstack
+    command: [entrypoint, prow/integ-suite-kind.sh, doc.test.dualstack]
+    requirements: [kind]
+    env:
+      - name: DOCKER_IN_DOCKER_IPV6_ENABLED
+        value: "true"
+      - name: IP_FAMILY
+        value: "dual"
+    regex: ^(go.mod$|tests/|pkg/test/|prow/|content/en/boilerplates/snips/|content/en/docs/.*(test\.sh|/snips\.sh)$)
```
Comment (ericvn) [inline prow/config/jobs/istio.io.yaml:52]: Let's also make this optional until we've done some testing with the new tests:
```suggestion
    regex: ^(go.mod$|tests/|pkg/test/|prow/|content/en/boilerplates/snips/|content/en/docs/.*(test\.sh|/snips\.sh)$)
    modifiers: [presubmit_optional]
```

**Review by ericvn — COMMENTED** [review]
You'll also need to run `make gen` to create the generated files.

**Comment by ericvn** [comment]
The change will require another `make gen` to put the optional in the generated files.

**Comment by kfaseela** [comment]
> The change will require another `make gen` to put the optional in the generated files.

Done

**Review by ericvn — APPROVED** [review]
This job is optional and should be merged to test the instio.io test. We may need to revisit and tweak as that test is updated.

**Pull request merged**